src/cli/self_update.rs

```diff
@@ -953,21 +953,20 @@ fn delete_rustup_and_cargo_home() -> Result<()> {
     // of CARGO_HOME.
     let numbah: u32 = rand::random();
     let gc_exe = work_path.join(&format!("rustup-gc-{:x}.exe", numbah));
+    // Copy rustup (probably this process's exe) to the gc exe
+    utils::copy_file(&rustup_path, &gc_exe)?;
+    let gc_exe_win: Vec<_> = gc_exe.as_os_str().encode_wide().chain(Some(0)).collect();
+
+    // Open an inheritable handle to the gc exe marked
+    // FILE_FLAG_DELETE_ON_CLOSE. This will be inherited
+    // by subsequent processes.
```

Comment:
I'd suggest the comment needs tweaking here -- the thing it comments is the inheritance marker, but the file flag is actually later in the unsafe code.